src/utilities/tuples.ts

```diff
@@ -0,0 +1,16 @@
+export type TupleType<T extends string[]> = {
+  tuple: T,
+  isTuple: (value: unknown) => value is T,
+}
+
+export function createTuple<const T extends string[]>(tuple: T): TupleType<T> {
+
+  function isTuple(value: unknown): value is T {
+    return tuple.includes(value as T[number])
+  }
```

Comment:
This function seems confusing - or atleast i'm confused about it's use. 

```ts
const { tuple, isTuple } = createTuple(['a', 'b', 'c'])

const someValue = 'b'
if (isTuple(someValue)) {
  console.log(someValue)  // on hover, TS shows someValue as `'b' & ['a', 'b', 'c']` so it thinks it has array-like methods on it now too
}
```

It reads like this tells me whether a value is valid _for_ the tuple like if you want to check existence against the tuple in an idiomatic way (but why not just use `tuple.includes`?). Or i'm vastly misunderstanding what the tuple itself is

Reply:
> but why not just use tuple.includes?

Because `tuple.includes` won't accept `unknown` values
<img width="832" alt="image" src="https://github.com/PrefectHQ/prefect-ui-library/assets/6200442/e46181d8-ddb8-427d-b9d8-3f9361ebd99b">